types/url-parse/index.d.ts

```diff
@@ -23,12 +23,12 @@ declare namespace URLParse {
         | 'slashes'
         | 'username';
 
-    type QueryParser = (query: string) => object;
+    type QueryParser<T = Record<string, string>> = (query: string) => T;
```

Comment:
`Record<string, string>` is like `any` in that it expects literally any key used on the `Record` to have type `string`. Since a given object probably does not have a `string` value for every possible key, it's best to use `Record<string, string | undefined>`.